### PR TITLE
Add eventarc trigger missing field event_data_content_type

### PR DESCRIPTION
**Pull request by RileyHYZ**
This PR upgrades dcl version to v.1.48.0 and adds a field event_data_content_type in eventarc trigger.

fixes https://github.com/hashicorp/terraform-provider-google/issues/14597

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [X] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
eventarc: added field `event_data_content_type` to `google_eventarc_trigger`
```

### CODE REVIEW

File: tpgtools/overrides/eventarc/samples/trigger/basic_update_json.tf.tmpl [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	matching_criteria {
+		attribute = "type"
+		value = "google.cloud.pubsub.topic.v1.messagePublished"
+	}
+	destination {
+		cloud_run_service {
+			service = google_cloud_run_service.default.name
+			region = "europe-west1"
+		}
+	}s
```
Comment (slevenick) [inline tpgtools/overrides/eventarc/samples/trigger/basic_update_json.tf.tmpl:13]: Typo here may cause test failures

**Comment by slevenick** [comment]
Can you try pushing a new commit? This got caught in the GitHub downtime

**Comment by adsko** [comment]
@slevenick @RileyHYZ Hey! Nice job with this one, but is there any update about this one? Would be nice to have it.

**Comment by slevenick** [comment]
The DCL was upgraded to 1.44.0 but this field doesn't seem to be present. I'm not sure why, as it exists in the YAML file for that resource in this release: https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blame/main/services/google/eventarc/trigger.yaml

**Comment by slevenick** [comment]
Ah I see what happened. It's present in GA but not beta, so things are a little weird. I'll ping someone to get this resolved

**Comment by RileyHYZ** [comment]
The change is intended for GA but not beta and the issue should have been resolved with dcl version>1.42.0. I'll resolve the conflict and commit again.

**Comment by adsko** [comment]
@slevenick

**Review by slevenick — CHANGES_REQUESTED** [review]
Looks like updating the DCL has introduced an issue in dataplex asset. Several tests are failing due to large diffs. I'll need to reach out to some people to see what is going on there

**Comment by NickElliot** [comment]
Update to resolve the issue on DCL's end is in review. Apologies for the complications!

**Comment by NickElliot** [comment]
The DCL issue should be resolved now with the new release.

**Review by slevenick — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/go.mod.erb [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.19
 
 require (
 	cloud.google.com/go/bigtable v1.19.0
-	github.com/GoogleCloudPlatform/declarative-resource-client-library v1.44.0
+	github.com/GoogleCloudPlatform/declarative-resource-client-library v1.47.0
```
[the remark this answers is not in the document]
Comment (RileyHYZ) [reply mmv1/third_party/terraform/go.mod.erb]: Sorry I didn't see the upgraded version earlier. Just updated it in the new commit. Thanks!

**Review by slevenick — APPROVED** [review]
Thanks!

**Pull request merged**